### PR TITLE
New rider frame with some new api added

### PR DESCRIPTION
**Pull request by StarryVae**

### CODE REVIEW

File: example/rider/rider_v2.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          http_filters:
+          - name: proxy.filters.http.rider
+            typed_config:
+              "@type": type.googleapis.com/proxy.filters.http.rider.v3alpha1.FilterConfig
+              plugin:
+                vm_config:
+                  package_path: "/home/starrynight/workspace/rider/?/init.lua;/home/starrynight/workspace/rider/?.lua;"
+                code:
+                  local:
+                    filename: /home/starrynight/workspace/rider/plugins/test/fun_test_v2.lua
```
Comment (wbpcode) [inline example/rider/rider_v2.yaml:47]: Please remove personal info in this example yaml

File: source/filters/http/rider/context.h [hunk 2]
```diff
@@ -72,13 +72,26 @@ class PluginHandle : public ThreadLocal::ThreadLocalObject {
   int& onRequestRef() { return on_request_ref_; }
   int& onResponseRef() { return on_response_ref_; }
 
+  int& onRequestHeaderRef() { return on_request_header_ref_; }
+  int& onRequestBodyRef() { return on_request_body_ref_; }
+  int& onResponseHeaderRef() { return on_response_header_ref_; }
+  int& onResponseBodyRef() { return on_response_body_ref_; }
+
+  std::string& version() { return version_; }
```
Comment (wbpcode) [inline source/filters/http/rider/context.h:80]: enum would be better for the version.

File: source/filters/http/rider/filter.cc [hunk 3]
```diff
@@ -154,41 +152,56 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap &headers,
...
   request_headers_ = &headers;
-  return doHeaders(
-      request_stream_wrapper_, StreamDirection::DownstreamToUpstream,
-      *(config_.pluginHandle().vm()), request_coroutine_, decoder_callbacks_,
-      config_.pluginHandle().onRequestRef(), headers, end_stream);
+  if (config_.pluginHandle().version().empty()) {
+    return doHeaders(request_stream_wrapper_, StreamDirection::DownstreamToUpstream,
+                     *(config_.pluginHandle().vm()), request_coroutine_, decoder_callbacks_,
+                     config_.pluginHandle().onRequestRef(),
+                     config_.pluginHandle().onRequestBodyRef(), headers, end_stream);
```
Comment (wbpcode) [inline source/filters/http/rider/filter.cc:168]: onRequestBodyRef?

**Review by wbpcode — CHANGES_REQUESTED** [review]
Thanks. And some minor comments.

File: example/rider/rider_v2.yaml (hunk 1 above)
Comment (StarryVae) [reply example/rider/rider_v2.yaml]: done

File: source/filters/http/rider/context.h (hunk 2 above)
Comment (StarryVae) [reply source/filters/http/rider/context.h]: done

File: source/filters/http/rider/filter.cc (hunk 3 above)
Comment (StarryVae) [reply source/filters/http/rider/filter.cc]: done

**Comment by wbpcode** [comment]
LGTM.

**Pull request merged**